### PR TITLE
adding base64 UIImage initialiser #741 

### PR DESCRIPTION
**Pull request by thisIsTheFoxe**
When you want to create an UIImage using a base 64 string.

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [?] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

?  (it compiled without errors or warnings without changing the deployment target so it should work right?)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - Added `enumerateMatches(in:options:range:using:)`, `matches(in:options:range:)`, `numberOfMatches(in:options:range:)`, `firstMatch(in:options:range:)`, `rangeOfFirstMatch(in:options:range:)`, `stringByReplacingMatches(in:options:range:withTemplate:)`, `replaceMatches(in:options:range:withTemplate:)`, which use `String` and `String.Range` in place of `NSString` and `NSRange` to make the calls Swifter. [#727](https://github.com/SwifterSwift/SwifterSwift/pull/727) by [guykogus](https://github.com/guykogus).
 - **UIImage**:
   - Added `withBackgroundColor(_:)` to specify a background color for a partially transparent image. [#721](https://github.com/SwifterSwift/SwifterSwift/pull/721) by [MaxHaertwig](https://github.com/maxhaertwig).
+  - Added `init?(base64String string: String)` to create an UIImage from a base64 String. [#741](https://github.com/SwifterSwift/SwifterSwift/issues/741) by [@thisIsTheFoxe](https://github.com/thisisthefoxe)
```
Comment (guykogus) [inline CHANGELOG.md]:
```suggestion
  - Added `init?(base64String:)` to create a `UIImage` from a base-64 `String`. [#741](https://github.com/SwifterSwift/SwifterSwift/issues/741) by [@thisIsTheFoxe](https://github.com/thisisthefoxe)
```

File: Sources/SwifterSwift/UIKit/UIImageExtensions.swift [hunk 2]
```diff
@@ -309,6 +309,15 @@ public extension UIImage {
         self.init(cgImage: aCgImage)
     }
 
+    /// SwifterSwift: Create a new image from a base 64 string.
+    ///
+    /// - Parameters:
+    ///   - string: a base 64 string, representing the image
+    convenience init?(base64String string: String) {
+        guard let data = Data(base64Encoded: string, options: .ignoreUnknownCharacters) else { return nil }
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UIImageExtensions.swift]: Why do you want to add `.ignoreUnknownCharacters`? It seems to me like this should fail if there are invalid characters included, no?

File: Tests/UIKitTests/UIImageExtensionsTests.swift [hunk 3]
```diff
@@ -123,6 +123,12 @@ final class UIImageExtensionsTests: XCTestCase {
         XCTAssertEqual(emptyImage, filledImage)
     }
 
+    func testBase64() {
+        let base64Str = "iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAE0lEQVR42mP8v5JhEwMaYKSBIADNAwvIr8dhZAAAAABJRU5ErkJggg=="
+        let img = UIImage(base64String: base64Str)
+        XCTAssertNotNil(img)
```
Comment (guykogus) [inline Tests/UIKitTests/UIImageExtensionsTests.swift]: Are there any more things you can test, such as the size of the image?

File: Tests/UIKitTests/UIImageExtensionsTests.swift [hunk 4]
```diff
@@ -123,6 +123,12 @@ final class UIImageExtensionsTests: XCTestCase {
         XCTAssertEqual(emptyImage, filledImage)
     }
 
+    func testBase64() {
+        let base64Str = "iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAE0lEQVR42mP8v5JhEwMaYKSBIADNAwvIr8dhZAAAAABJRU5ErkJggg=="
```
Comment (guykogus) [inline Tests/UIKitTests/UIImageExtensionsTests.swift]: Use full names
```suggestion
        let base64String = "iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAE0lEQVR42mP8v5JhEwMaYKSBIADNAwvIr8dhZAAAAABJRU5ErkJggg=="
```

File: Tests/UIKitTests/UIImageExtensionsTests.swift [hunk 5]
```diff
@@ -123,6 +123,12 @@ final class UIImageExtensionsTests: XCTestCase {
...
+        let img = UIImage(base64String: base64Str)
```
Comment (guykogus) [inline Tests/UIKitTests/UIImageExtensionsTests.swift]: Same as above
```suggestion
        let image = UIImage(base64String: base64Str)
```

File: Sources/SwifterSwift/UIKit/UIImageExtensions.swift (hunk 2 above)
Comment (thisIsTheFoxe) [reply Sources/SwifterSwift/UIKit/UIImageExtensions.swift]: I also was unsure about this.. Will remove it, thanks for the hint ^^

File: Tests/UIKitTests/UIImageExtensionsTests.swift [hunk 6]
```diff
@@ -123,6 +123,17 @@ final class UIImageExtensionsTests: XCTestCase {
         XCTAssertEqual(emptyImage, filledImage)
     }
 
+    func testBase64() {
+        let base64String = "iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAE0lEQVR42mP8v5JhEwMaYKSBIADNAwvIr8dhZAAAAABJRU5ErkJggg=="
+        let image = UIImage(base64String: base64String)
+        XCTAssertNotNil(image)
+
+        let size: CGSize = CGSize(width: 5, height: 5)
```
Comment (guykogus) [inline Tests/UIKitTests/UIImageExtensionsTests.swift]:
```suggestion
        let size = CGSize(width: 5, height: 5)
```

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃

**Comment by thisIsTheFoxe** [comment]
Cool

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**